### ISSUE TITLE
[Mod Support] Update Abnormals Delight Support

### ISSUE THREAD

**Issue by NJL5390** [TheIllusiveC4/Diet#83]
Hi, 

Abnormals Delight: (https://www.curseforge.com/minecraft/mc-mods/abnormals-delight) Recently got updated, Adding a few new foods. 
 Below you can find how I've categorised the new foods added in this update:

Fruits:
    "abnormals_delight:cherry_cookie",
    "abnormals_delight:mulberry_cookie",
    "abnormals_delight:cherry_cream_soda",
    "abnormals_delight:passion_aloe_nectar"

Grains:
    "abnormals_delight:duck_noodles",
    "abnormals_delight:cherry_cookie",
    "abnormals_delight:maple_cookie",
    "abnormals_delight:mulberry_cookie"

Proteins:
    "abnormals_delight:duck_noodles",
    "abnormals_delight:perch_with_mushrooms",
    "abnormals_delight:pike_with_beetroot",
    "abnormals_delight:venison_with_bamboo_shoots",
    "abnormals_delight:maple_glazed_bacon",
    "abnormals_delight:slabdish"

Sugars:
    "abnormals_delight:maple_cookie",
    "abnormals_delight:maple_glazed_bacon",
    "abnormals_delight:cherry_cream_soda",
    "abnormals_delight:passion_aloe_nectar",
    "abnormals_delight:pickerelweed_juice"

Vegetables:
    "abnormals_delight:duck_noodles",
    "abnormals_delight:perch_with_mushrooms",
    "abnormals_delight:pike_with_beetroot",
    "abnormals_delight:venison_with_bamboo_shoots",
    "abnormals_delight:slabdish",
    "abnormals_delight:pickerelweed_juice",
    "abnormals_delight:passion_aloe_nectar"

As always, Thanks a lot and keep up the great work!

**Comment by NielsPilgaard** [comment]
`abnormals_delight:venison_with_bamboo_shoots` seems to be missing the protein tag still.
https://github.com/EnigmaticaModpacks/Enigmatica6/issues/4417

**Comment by TheIllusiveC4** [comment]
Hmm, I wonder if that's probably a side-effect of a big revamp I did for Diet a while ago that automatically generates the values. I had to delete a lot of the hardcoded ones and some foods may have gotten lost in the shuffle. I'll look into this again, let me know if there's any other Abnormal's Delight food that has missing values.

**Comment by NielsPilgaard** [comment]
Will do, thanks 👍

**Comment by TheIllusiveC4** [comment]
Okay, so I found out something weird while investigating this. With just Diet, Environmental, Farmer's Delight, and Abnormal's Delight installed, the `abnormals_delight:venison_with_bamboo_shoots` has the protein tag just fine. However, testing this with E6E as reported by the issue tracker on Enigmatica, I see that it is in fact missing.

So why the discrepancy? Well, I think it might have something to do with the fact that the modpack does not use the raw venison shanks from Abnormal's Delights. It is missing from JEI search and the recipe for the Venison with Bamboo Shoots does not use it as an ingredient. However, it is still able to be found in Creative search and I can see that the venison shanks do not have a protein tag. What I think is happening is that Diet is using the venison shanks as the base for the recipe instead of the regular venison, which causes the protein tag to be missing from the recipe result because it is missing from the venison shanks. Note that in a minimal mod environment, the venison shanks do have the protein tag.

@NillerMedDild Do you happen to know how you're removing the venison shanks and if you are removing the protein tag from it as well? Relevant links to any source code/scripts would be helpful as well. I just need some more information about the relevant changes the modpack may have made to these items in order to better investigate how I could best address this.

I could simply just add the item to the proper tags and call it a day, but I'd like to see if it's a deeper issue with how Diet is processing recipes or items that may have been edited in a modpack environment.

**Comment by NielsPilgaard** [comment]
> @NillerMedDild Do you happen to know how you're removing the venison shanks and if you are removing the protein tag from it as well? Relevant links to any source code/scripts would be helpful as well. I just need some more information about the relevant changes the modpack may have made to these items in order to better investigate how I could best address this.

We remove all recipes where Venison Shanks is the output, hide it from JEI, and add a description saying it's disabled. 
Here's where we remove all recipes where it's the output (It's part of the `disabledItems` array): https://github.com/EnigmaticaModpacks/Enigmatica6/blob/master/kubejs/server_scripts/enigmatica/kubejs/base/recipes/remove.js#L281

Here's where we hide it from JEI: 
https://github.com/EnigmaticaModpacks/Enigmatica6/blob/master/kubejs/client_scripts/item_modifiers/jei_hide_items.js#L33

The meat used in stead of Venison Shanks is Raw Venison, and it does have the protein tag.

Please let me know if you need any other info :)

**Comment by TheIllusiveC4** [comment]
Sorry this took a while. What happened is that I discovered some more discrepancies in how the food groups were being calculated so I bundled this issue with another one where I just revamped the whole algorithm and that took a bit of time to do all the right testing.

This particular issue for that particular pack should be solved but because of the new algorithm other items may have new food groups that they didn't before. In 99% of cases it should be the same as before, but it's still something to be aware of. Report any discrepancies to the issue tracker and I'll look into it.
